heat/core/_operations.py

```diff
@@ -111,28 +110,31 @@ def __binary_op(
             output_device = t1.device
             output_comm = t1.comm
 
-            # ToDo: Fine tuning in case of comm.size>t1.shape[t1.split]. Send torch tensors only to ranks, that will hold data.
             if t1.split is not None:
                 if t1.shape[t1.split] == 1 and t1.comm.is_distributed():
                     # warnings.warn(
                     #     "Broadcasting requires transferring data of first operator between MPI ranks!"
                     # )
-                    if t1.comm.rank > 0:
+                    color = 0 if t1.comm.rank < t2.shape[t1.split] else 1
+                    newcomm = t1.comm.Split(color, t1.comm.rank)
+                    if t1.comm.rank > 0 and color == 0:
                         t1.larray = torch.zeros(
                             t1.shape, dtype=t1.dtype.torch_type(), device=t1.device.torch_device
                         )
-                    t1.comm.Bcast(t1)
+                    newcomm.Bcast(t1)
```

Comment:
i dont see why we need a split. can you explain that to me?

Also, splits can cause a little bit of overhead in the general code. If you are splitting for this one communication, it might have other effects down the line

Reply:
I address the problem described by comment in line 114. At the moment rank 0 broadcasts the data of an operand to all other processes causing problems as seen in the linked issue. I split the communicator such that the broadcast from rank 0 is now restricted to the ones that will need the data. On the other ranks an empty array is broadcast leaving them unchanged.

Reply:
doing this in the abstracted operation can cause performance issues. lets say that you are doing a lot of arithmetic operations where this is the case, then you would be creating a new mpi split comm for each one. they are not cleaned up by the normal garbage collector, so if we do it this way, the new communicator needs to be Finalized and removed from the MPI runtime.

Reply:
Would a call to comm.Free() be enough?

Reply:
should be